bindings/cpp/src/Session.hpp

```diff
@@ -208,6 +209,14 @@ class Subscribe
 
     S_Session sess;
     S_Deleter sess_deleter;
+
+    bool uses_custom_loop;
```

Comment:
The "with event loop" constructor takes a pair of two `std::functions`, which will get called later on, but their validity is not checked. I propose to check that they are defined, and use that in place of `uses_custom_loop`, it's gonna be simpler that way.

Reply:
done